Model/SourcesRepository.php

```diff
@@ -0,0 +1,178 @@
+     * @param SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory
+     * @param SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+     */
+    public function __construct(
+        SourcesFactory $sourcesFactory,
+        Sources $sourcesResourceModel,
+        SerializerInterface $serializer,
+        \Magento\Framework\Api\DataObjectHelper $dataObjectHelper,
+        SourceSelectionResultInterfaceFactory $sourceSelectionResultFactory,
+        SourceSelectionItemInterfaceFactory $sourceSelectionItemInterface
+    ) {
+        $this->sourcesFactory = $sourcesFactory;
+        $this->sourcesResourceModel = $sourcesResourceModel;
+        $this->serializer = $serializer;
+        $this->dataObjectHelper = $dataObjectHelper;
+        $this->sourceSelectionResultFactory = $sourceSelectionResultFactory;
+        $this->sourceSelectionItemInterface = $sourceSelectionItemInterface;
+
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws NoSuchEntityException
+     */
+    public function getByOrderId(string $orderId): SourceModel
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        return $sourcesModel;
+    }
+
+    /**
+     * @param string $orderId
+     *
+     * @return SourceSelectionResultInterface
+     */
+    public function getSourceSelectionResultByOrderId(string $orderId): SourceSelectionResultInterface
+    {
+        /** @var SourceModel $sourcesModel */
+        $sourcesModel = $this->sourcesFactory->create();
+
+        $this->sourcesResourceModel->load(
+            $sourcesModel,
+            $orderId,
+            'order_id'
+        );
+
+        $sourcesArray = $this->serializer->unserialize($sourcesModel->getSources());
+        $sourceSelectionItems = [];
+
+        foreach ($sourcesArray as $item)
+        {
+            $sourceSelectionItem = $this->sourceSelectionItemInterface->create(
+                [
+                    'sourceCode' => $item['source_code'],
+                    'sku' => $item['SKU'],
+                    'qtyToDeduct' => $item['qty_to_deduct'],
+                    'qtyAvailable' => $item['qty_available']
+                ]
+            );
+
+        $sourceSelectionItems[] = $sourceSelectionItem;
+        }
+
+        $sourceSelectionResult = $this->sourceSelectionResultFactory->create(
+            [
+                'sourceItemSelections' => $sourceSelectionItems,
+                'isShippable' => true
+            ]
+        );
+
+        return $sourceSelectionResult;
+    }
+
+    /**
+     * @param array $sourcesItems
+     * @param string $orderId
+     *
+     * @return SourceModel
+     * @throws CouldNotSaveException
+     */
+    public function save(array $sourcesItems, string $orderId): SourceModel
```

Comment:
Usually the repository method would take an instance of the model and then save it, for example https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Cms/Model/BlockRepository.php#L112 ... I think we should do something similar and then move the conversion logic into its own classes/methods?

Reply:
@ahasAmp Again, I think this should take a source selection model and save it, rather than an array of items and an order ID. Does that make sense? So you'd be passing this function a `SourceModel`